src/cargo/core/compiler/custom_build.rs

```diff
@@ -397,35 +399,63 @@ fn build_work(cx: &mut Context<'_, '_>, unit: &Unit) -> CargoResult<Job> {
         let timestamp = paths::set_invocation_time(&script_run_dir)?;
         let prefix = format!("[{} {}] ", id.name(), id.version());
         let mut diagnostics = Vec::new();
-        let output = cmd
-            .exec_with_streaming(
-                &mut |stdout| {
-                    if let Some(warning) = stdout.strip_prefix(CARGO_WARNING) {
-                        diagnostics.push(Diagnostic::Warning(warning.to_owned()));
-                    } else if let Some(error) = stdout.strip_prefix(CARGO_ERROR) {
-                        diagnostics.push(Diagnostic::Error(error.to_owned()));
-                    }
-                    if extra_verbose {
-                        state.stdout(format!("{}{}", prefix, stdout))?;
-                    }
-                    Ok(())
-                },
-                &mut |stderr| {
-                    if extra_verbose {
-                        state.stderr(format!("{}{}", prefix, stderr))?;
-                    }
-                    Ok(())
-                },
-                true,
-            )
-            .with_context(|| format!("failed to run custom build command for `{}`", pkg_descr));
```

Comment:
Is it a problem that we are no longer mentioning which package's build script failed?  Its hard to tell from the one test case covered

Reply:
Don't know how i missed that in the test.